rasa/nlu/featurizers/sparse_featurizer/regex_featurizer.py

```diff
@@ -45,58 +52,67 @@ def train(
         self._add_lookup_table_regexes(training_data.lookup_tables)
 
         for example in training_data.training_examples:
-            updated = self._text_features_with_regex(example)
-            example.set(MESSAGE_VECTOR_FEATURE_NAMES[MESSAGE_TEXT_ATTRIBUTE], updated)
+            self._text_features_with_regex(example)
 
     def process(self, message: Message, **kwargs: Any) -> None:
+        self._text_features_with_regex(message)
 
-        updated = self._text_features_with_regex(message)
-        message.set(MESSAGE_VECTOR_FEATURE_NAMES[MESSAGE_TEXT_ATTRIBUTE], updated)
-
-    def _text_features_with_regex(self, message):
+    def _text_features_with_regex(self, message: Message) -> None:
         if self.known_patterns:
             extras = self.features_for_patterns(message)
-            return self._combine_with_existing_features(message, extras)
+            features = self._combine_with_existing_sparse_features(message, extras)
         else:
-            return message.get(MESSAGE_VECTOR_FEATURE_NAMES[MESSAGE_TEXT_ATTRIBUTE])
+            features = message.get(
+                MESSAGE_VECTOR_SPARSE_FEATURE_NAMES[MESSAGE_TEXT_ATTRIBUTE]
+            )
+
+        message.set(
+            MESSAGE_VECTOR_SPARSE_FEATURE_NAMES[MESSAGE_TEXT_ATTRIBUTE], features
+        )
 
-    def _add_lookup_table_regexes(self, lookup_tables):
+    def _add_lookup_table_regexes(
+        self, lookup_tables: List[Dict[Text, Union[Text, List]]]
+    ) -> None:
         # appends the regex features from the lookup tables to
         # self.known_patterns
         for table in lookup_tables:
             regex_pattern = self._generate_lookup_regex(table)
             lookup_regex = {"name": table["name"], "pattern": regex_pattern}
             self.known_patterns.append(lookup_regex)
 
-    def features_for_patterns(self, message):
+    def features_for_patterns(self, message: Message) -> scipy.sparse.csr_matrix:
```

Comment:
```suggestion
    def _features_for_patterns(self, message: Message) -> scipy.sparse.csr_matrix:
```
why it is not private?